### ISSUE TITLE
Is Aim 1.1 done?

### ISSUE THREAD

**Issue by brandonjnelsonFDA** [bnel1201/PedSilicoICH#69]
# Background
![image](https://github.com/user-attachments/assets/44807e93-b1f5-44b5-82f1-c99195c08ce6)

This aim is on characterizing our real datasets, and we focused at first on contrast/intensity, volume, and lesion type (IPH, SDH, EPH) distributions among the three public datasets we sourced:

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 15">
<meta name=Originator content="Microsoft Word 15">
<link rel=File-List
href="file:///C:/Users/BRANDO~1.NEL/AppData/Local/Temp/1/msohtmlclip1/01/clip_filelist.xml">
<link rel=Preview
href="file:///C:/Users/BRANDO~1.NEL/AppData/Local/Temp/1/msohtmlclip1/01/clip_preview.wmf">

<link rel=themeData
href="file:///C:/Users/BRANDO~1.NEL/AppData/Local/Temp/1/msohtmlclip1/01/clip_themedata.thmx">
<link rel=colorSchemeMapping
href="file:///C:/Users/BRANDO~1.NEL/AppData/Local/Temp/1/msohtmlclip1/01/clip_colorschememapping.xml">

<style>

</style>

</head>

<body lang=EN-US link=blue vlink=purple style='tab-interval:.5in;word-wrap:
break-word'>

<div align=center>

Dataset | Cases | Includes Peds | Age provided | Segmentations | ICH Type
-- | -- | -- | -- | -- | --
Instance | 130 | no | no | yes | no
RSNA | 192 | no | no | yes | yes
Hssayeni | 82 | yes | yes | yes | yes

</div>

</body>

</html>

I often use this quick chart to show the overlaid distributions of 2/3 datasets (still need to add Hssayeni) and add another subplot for lesion type distributions (SDH, EDH, IPH, control...)

![image](https://github.com/user-attachments/assets/f08382af-3354-4766-aaba-7da602cc22e4)

I see we have 4 notebooks in this repo that all seem related to aim 1.1:

- [RSNA_BHSD_explore](https://github.com/DIDSR/PedSilicoICH/blob/master/notebooks/RSNA_BHSD_explore.ipynb)
- [RSNA_BHSD_fitting](https://github.com/DIDSR/PedSilicoICH/blob/master/notebooks/RSNA_BHSD_fitting.ipynb)
- [Instance 2022 explore](https://github.com/DIDSR/PedSilicoICH/blob/master/notebooks/instance2022_explore.ipynb)
- [Hyssayeni explore](https://github.com/DIDSR/PedSilicoICH/blob/master/notebooks/00_basic_eda.ipynb)

Here's our 6 month skynet study report for reference: [study report 01.pdf](https://github.com/user-attachments/files/17664230/study.report.01.pdf)

# Proposed Task

I believe editing and merging these notebooks down to a single Aim 1.1 notebook that outputs an aim 1.1 figure for the paper and some of the [fit distributions](https://github.com/DIDSR/PedSilicoICH/tree/master/src/pedsilicoICH/distributions) for the synthetic data modeling may be helpful to close the loop here.

**Comment by jmweaver-FDA** [comment]
I agree that Aim 1.1 is essentially as done as it can be without finding/including more datasets - a summary notebook will definitely be helpful, and I can get to work on that. There really isn't a great way t compare distributions between datasets, especially as INSTANCE and Hssayeni don't have individual hemorrhage delineation - volumes from Instance and Hssayeni will likely always be higher than in my BHSD analysis.

I'll start working on an Aim 1.1 summary notebook that mostly focuses on high-quality figures for use in future publications and presentations!

**Comment by brandonjnelsonFDA** [comment]
I agree we don't need additional datasets, but am leaning towards a final summary notebook and any data cleaning of our existing datasets that may help in doing comparisons between datasets. And thank you for volunteering, we can work together on it too, just let me know your bandwidth.

For example for Instance and Hssayeni, if they are semantic segmentations and not instance segmentations, perhaps we could use [scikit-images label function](https://scikit-image.org/docs/stable/api/skimage.measure.html#skimage.measure.label) to turn the disconnected segmentations into separately labeled hemorrhages like BHSD? - See [scikit-image segmentation examples](https://scikit-image.org/docs/stable/auto_examples/segmentation/plot_label.html#sphx-glr-auto-examples-segmentation-plot-label-py)

![image](https://github.com/user-attachments/assets/f1595f59-836a-45bd-86a4-7075b144ffc7)

**Comment by brandonjnelsonFDA** [comment]
A side thought, we've been discussing with Rucha about adding shape analysis, I see [sckit-image has functions for shape analysis, including curvature](https://scikit-image.org/docs/stable/auto_examples/features_detection/plot_shape_index.html#sphx-glr-auto-examples-features-detection-plot-shape-index-py), wonder if this is a simple way to go

**Comment by jmweaver-FDA** [comment]
We can absolutely use `label` on both datasets similar to what I did with BHSD, we just don't be able to differentiate based on lesion type because that isn't separately labeled. However, we could aggregate all hemorrhages together (agnostic of lesion type) and look at distributions there. 

However, we do know from the RSNA_BHSD_explore notebook that different hemorrhage types do have different distributions (particularly noticeable with hemorrhage HU).

**Comment by brandonjnelsonFDA** [comment]
Also these datasets are small enough and we have worked with then long enough that we can label them manually when lesion type annotations arent there, so long as we are systematic: away from skull = iph, on skull+concave = sdh, ...

We can list those details in an appendix/lab notebook and we'll get radiologist confirmation or corrections on any of our manual annotations when we do our informal image review session

**Comment by jmweaver-FDA** [comment]
Agreed - it may be worthwhile to cook up a very simple Jupyter notebook that can be used to annotate instead of some external GUI that we could easily send to someone else for radiologist confirmation. Something like the following workflow:

1. Load specific case
2. Use scikit label function to segment unique hemorrhage
3. Looping through each unique hemorrhage, show in image viewer with mask, w/ dropdown and label
4. Repeat 1-3 for next case

Results could be saved in a CSV that contains duplicate decisions from the two of us and others until a consensus is reached. 

Honestly, if we did this for both Instance and Hssayeni we could in theory combine them with BHSD into a super dataset that could then be released (albeit with the original authors permission).... it would probably be extremely beneficial for the segmentation community to have a large dataset with type-specific segmentations

**Comment by brandonjnelsonFDA** [comment]
1. I agree with your annotation plan, your use of widgets for labeling in step 3 is elegant, likely more reproducible and more transparent - all things i like if its not too difficult to implement.

Alternatively, you could also just dump the outputs from 'label' into a csv with x, y, z coords and we can manually label lesion type in Excel. Your call, either way.

2. Improving datasets and republishing them is a great service to the community and totally worth doing, I'm all for it. we recently did this in DIDSR by cleaning up pediatric bone age labels with the help of a passionate radiologist, similarly isnt BHSD an example of this? Improving the label quality of (a subset) of the RSNA dataset?

**Comment by jmweaver-FDA** [comment]
Yes - I'm proposing we do exactly what the BHSD group did with the RSNA dataset to both Hssayeni and INSTANCE!

**Comment by brandonjnelsonFDA** [comment]
Awesome, if you can get started on a pilot (or delegate in a separate linked issue) and if it looks good we can add the improved real dataset as another project output 🏆

**Comment by jmweaver-FDA** [comment]
Update: a new [notebook](https://github.com/DIDSR/PedSilicoICH/blob/0084a0352a2e3dcf73a07f43348af4dab5dbaa74/notebooks/Aim_1point1.ipynb) has been created to aggregate the data from all three datasets into one [master spreadsheet](https://github.com/DIDSR/PedSilicoICH/blob/0084a0352a2e3dcf73a07f43348af4dab5dbaa74/notebooks/MASTER_DATASET.csv).

As discussed above, the spreadsheet currently has hemorrhage types for BHSD (as provided) but hemorrhage type is currently "UNKNOWN" for Hssayeni and Instance. Future work will involve classifying these hemorrhages.

This issue could remain open until some publication-quality figures are added to the Aim 1.1 Notebook, or until the above classifications are complete.

**Comment by brandonjnelsonFDA** [comment]
This summary notebook is fantastic! It looks like you were able to use scikit-image's label function successfully in cells 3, 4! From the subtle details of listing run times and summarizing the purpose of the notebook and ending with summary figures:

![image](https://github.com/user-attachments/assets/d5576333-c2ed-4427-86ff-42d99e5d5a86)

Your conclusion cell is nice, you make good points about differences in lesion type or age that could cause subtle shifts in HU distribution, I also argue that given the small size of each of these datasets the uncertainty in distributional mean HU of +-5 HU is to be expected.

I think this Aim 1.1 figure is very nearly there, it will become clearer as the draft nears completion. I could imagine a 4th KDE line for HU and lesion vol plots for joint distribution (i.e the mean across the 3 real datasets) that is used to inform the synthetic data distribution. A later plot could also compare the joint real distribution and synthetic data distribution, but that may be redundant as they are defined to be the same, but these are details we can work out when we get there!

![image](https://github.com/user-attachments/assets/6690cf68-4d20-44c7-b9af-5920723921f2)

Point is, great work and let's close this issue here and I'll open up a new issue classifying hemorrhage types that references this issue.

**Comment by jmweaver-FDA** [comment]
I'll add just one thing to this closed issue:

At the time of writing this (11/12/2024), the synthetic data distribution is defined _exclusively_ from the BHSD distributions obtained from the unique hemorrhage types, so the proposed edit to the above figure with joint distributions is not currently true of the synthetic data we have created thus far. 

Such a proposed figure would need to be created after Issue #70 is completed and we have hemorrhage-type distributions obtained from the use of all three datasets.

**Comment by brandonjnelsonFDA** [comment]
This is a good project management example of where the [github project backlog ](https://github.com/orgs/DIDSR/projects/22/views/4)can be used, if you think updating the synthetic data distributions with the joint distributions is worth doing and worth having a dedicated issue, you can make the issue and assign a start date to be right after #70 is completed (I put an arbitrary end date of 12/19 which can also be updated)

![image](https://github.com/user-attachments/assets/316c95f3-d0e1-4f32-a949-55b9b469d04f)
